### DISCUSSION TITLE
React-select & creatable select with reatc-hook-form focus on validation error

### DISCUSSION THREAD

**Discussion by sanathko** [react-hook-form/react-hook-form#5244] [Q&A] [answered]
I am using create select components inside a react-hook-form. I have tried the react-select and creatable-react-select both and the behaviour I experience in context of this issue is the same.

Using latest react-hook-form (7.6.1), I can't set the field to be focussed when there's an error bound to the field. 

https://codesandbox.io/s/react-hook-form-react-select-controller-with-focus-forked-cqsts?file=/src/index.js

When I use an older version of react-hook-form (5.4.2), I can get this behaviour to work as expected. 

https://codesandbox.io/s/react-hook-form-react-select-controller-with-focus-forked-ys4r4?file=/src/index.js

As i want this behaviour to work in the latest react-hook-form, any suggestions on what is the problem in the first example? and how can I get it sorted as per the behaviour using the old react-hook-form?

**Comment by leapful** [comment, marked as answer]
@sanathko Please provide the CSB to get more detail about the issue. Your current CSB is using react-hook-form v5.

**Reply by sanathko** [reply]
https://codesandbox.io/s/react-hook-form-react-select-controller-with-focus-forked-cqsts?file=/src/index.js using latest version of react hook form

**Reply by sanathko** [reply]
steps to replicate -  input box is required so if no value entered in input box, it will have the focus. When a value is entered in the inputbox and submit is pressed, the focus has to move to the next required field which is the select-input.

**Reply by leapful** [reply]
@sanathko 
 
You are using Controller for react-select component but did not use prop "field" from "render" method of Controller, therefore, the ref element is not set in order to help react-hook-form to set focus.

Working sample: https://codesandbox.io/s/react-hook-form-react-select-controller-with-focus-forked-9gb9t?file=/src/index.js
